db/db.go

```diff
@@ -50,6 +51,13 @@ const (
 // DB is the main interface for interacting with the
 // DefraDB storage system.
 type db struct {
+	// The ID of the last transaction created.
+	//
+	// Warning: we currently rely on this prop being 64-bit aligned in memory
+	// relative to the start of the `db` struct (for atomic.Add calls).  The
+	// easiest way to ensure this alignment is to declare it at the top.
+	previousTxnID uint64
```

Comment:
todo: go 1.19 introduced new atomic type to help make these variables safer. In this case I would encourage using `atomic.Uint64`.

Reply:
You can also remove the warning as the new type cover for 32bit systems
> [Int64](https://go.dev/pkg/sync/atomic/#Int64) and [Uint64](https://go.dev/pkg/sync/atomic/#Uint64) are automatically aligned to 64-bit boundaries in structs and allocated data, even on 32-bit systems.